test/dfusionMultiSafes.js

```diff
@@ -331,6 +341,106 @@ contract("GnosisSafe", function(accounts) {
       }
       await testAutomaticDeposits(tradeInfo, expectedDistribution)
     })
+    describe("can use automatic deposits to transfer tokens with arbitrary amount of decimals", () => {
+      const tokenSetups = [
+        {
+          amountStableToken: "10000",
+          amountTargetToken: "100",
+          stableTokenInfo: { decimals: 6, symbol: "USDC" },
+          targetTokenInfo: { decimals: 18, symbol: "WETH" },
+        },
+        {
+          amountStableToken: "100",
+          amountTargetToken: "10000",
+          stableTokenInfo: { decimals: 18, symbol: "WETH" },
+          targetTokenInfo: { decimals: 6, symbol: "USDC" },
+        },
+        {
+          amountStableToken: "3333",
+          amountTargetToken: "100.000001",
+          stableTokenInfo: { decimals: 0, symbol: "nodecimals" },
+          targetTokenInfo: { decimals: 6, symbol: "USDC" },
+        },
+        {
+          amountStableToken: "0.00000000000000000000001",
+          amountTargetToken: "3.14159265",
+          stableTokenInfo: { decimals: 100, symbol: "manydecimals" },
+          targetTokenInfo: { decimals: 8, symbol: "WBTC" },
+        },
+      ]
+      it("when p is in the middle of the brackets", async () => {
+        const tradeInfoWithoutTokens = {
+          fleetSize: 4,
+          lowestLimit: 100,
+          highestLimit: 121,
+          currentPrice: 110,
+        }
+        const expectedDistribution = {
+          bracketsWithStableTokenDeposit: 2,
+          bracketsWithTargetTokenDeposit: 2,
+        }
+        for (const tokenSetup of tokenSetups) {
+          const tradeInfo = {}
+          for (const key in tradeInfoWithoutTokens) tradeInfo[key] = tradeInfoWithoutTokens[key]
+          for (const key in tokenSetup) tradeInfo[key] = tokenSetup[key]
+          await testAutomaticDeposits(tradeInfo, expectedDistribution)
+        }
+      })
+      it("when p is not in the middle but still inside the brackets", async () => {
+        const tradeInfoWithoutTokens = {
+          fleetSize: 8,
+          lowestLimit: 100,
+          highestLimit: 130,
+          currentPrice: 110,
+        }
+        const expectedDistribution = {
+          bracketsWithStableTokenDeposit: 3,
+          bracketsWithTargetTokenDeposit: 5,
+        }
+        for (const tokenSetup of tokenSetups) {
+          const tradeInfo = {}
+          for (const key in tradeInfoWithoutTokens) tradeInfo[key] = tradeInfoWithoutTokens[key]
```

Comment:
Doesn't tradeInfo = tradeInfoWithouTokens work as well? There should be some copy command that does not require a for loop :)

Reply:
I couldn't find a copy command, not even for a shallow copy! If you know of one I'm happy to use it.

Reply:
I changed it. But I can't argue that it is strictly better than your solution.
Up to you, whether you want to keep it ... :)